src/main/java/com/razorpay/OAuthTokenClient.java

```diff
@@ -0,0 +1,126 @@
+package com.razorpay;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class OAuthTokenClient extends ApiClient {
+
+    static final String CLIENT_ID = "client_id";
+    static final String CLIENT_SECRET = "client_secret";
+    static final String GRANT_TYPE = "grant_type";
+    static final String REFRESH_TOKEN = "refresh_token";
+    static final String TOKEN = "token";
+    static final String TOKEN_TYPE_HINT = "token_type_hint";
+    static final String REDIRECT_URI = "redirect_uri";
+    static final String SCOPES = "scopes";
+    static final String STATE = "state";
+    static final String MODE = "mode";
+    private final PayloadValidator payloadValidator;
+
+    OAuthTokenClient(String auth) {
+        super(auth);
+        this.payloadValidator = new PayloadValidator();
+    }
+
+    public String getAuthURL(JSONObject request) throws RazorpayException {
+        validateAuthURLRequest(request);
+
+        String clientId = request.getString("client_id");
+        String redirectUri = request.getString("redirect_uri");
+        String state = request.getString("state");
+        String[] scopes = jsonArrayToStringArray(request.getJSONArray("scopes"));
+
+        String scopesArray = scopes.length > 0 ?
+                "&scope[]=" + String.join("&scope[]=", scopes) : "";
+
+        String AuthorizeUrl = "https://"
+                + Constants.AUTH_HOSTNAME
+                + "/" + Constants.AUTHORIZE
+                + "?response_type=code"
+                + "&client_id=" + clientId
+                + "&redirect_uri=" + redirectUri
+                + scopesArray
+                + "&state=" + state;
+
+        return AuthorizeUrl;
+    }
+
+    public OauthToken getAccessToken(JSONObject request) throws RazorpayException {
+        validateAccessTokenRequest(request);
```

Comment:
here the grant type will be authorization_code?

Reply:
Grant type can be authorization_code or client_credentials